Makefile

```diff
@@ -143,73 +157,70 @@ build-test-unit-js:
 # seems like the most straightforward way of capturing these
 # dependencies at the moment.
+	@echo "--- Building e2e pex files"
 	./pants --tag="e2e-test-pex" package ::
-	$(DOCKER_BUILDX_BAKE) \
-		--file ./test/docker-compose.integration-tests.build.yml \
-		e2e-tests
 
 .PHONY: build-engagement-view
-build-engagement-view: ## Build website assets
+build-engagement-view: ## Build website assets to include in grapl-web-ui
 	@echo "--- Building the engagement view"
 	$(MAKE) -C src/js/engagement_view build
 	cp -r \
 		"${PWD}/src/js/engagement_view/build/." \
 		"${PWD}/src/rust/grapl-web-ui/frontend/"
 
-.PHONY: build-ui
-build-ui: ## Build the web UI
-build-ui: build-engagement-view
-	@echo "--- Building the Web UI"
-	$(DOCKER_BUILDX_BAKE) --file docker-compose.build.yml \
-		grapl-web-ui
-
-.PHONY: build-grapl-services
-build-grapl-services: ## Build all the services for the Grapl SaaS
-build-grapl-services: build-ui
-	@echo "--- Building the rest of the Grapl services"
-# Our Python images need their PEX files built first
-	./pants --tag="service-pex" package ::
-	$(DOCKER_BUILDX_BAKE) --file docker-compose.build.yml \
-		analyzer-dispatcher \
-		analyzer-executor \
-		engagement-creator \
-		graph-merger \
-		graphql-endpoint \
-		model-plugin-deployer \
-		node-identifier \
-		node-identifier-retry \
-		osquery-generator \
-		plugin-bootstrap \
-		plugin-registry \
-		plugin-work-queue \
-		provisioner \
-		sysmon-generator
-
-.PHONY: build-local-services
-build-local-services: ## Build images used only in a local testing context
-	@echo "--- Building local-only services"
-	$(DOCKER_BUILDX_BAKE) --file docker-compose.build.yml \
-		localstack \
-		postgres \
-		pulumi
+.PHONY: build-grapl-service-prerequisites
+# The Python services need the PEX files
+build-grapl-service-prerequisites: build-service-pex-files
+# The grapl-web-ui service needs website assets
+build-grapl-service-prerequisites: build-engagement-view
+
+# This is used in our CI pipeline; see build_and_upload_containers.sh
+#
+# Also see the `push-to-cloudsmith` group in docker-bake.hcl; any
+# prerequisites of images in that group should be built by this
+# target!
+.PHONY: build-image-prerequisites
+build-image-prerequisites: ## Build all dependencies that must be copied into our images that we push to our registry
+build-image-prerequisites: build-grapl-service-prerequisites build-e2e-pex-files
+
+.PHONY: build-test-e2e
+build-test-e2e: build-e2e-pex-files
+	@echo "--- Building e2e testing image"
+	$(DOCKER_BUILDX_BAKE) --file "${BUILDX_BAKE_HCL_FILE}" e2e-tests
 
-.PHONY: build-for-push
-build-for-push: ## Build only the images we push to our registry
-build-for-push: build-grapl-services build-test-e2e
+.PHONY: build-test-integration
+build-test-integration:
+	@echo "--- Building integration test images"
+	$(DOCKER_BUILDX_BAKE) --file "${BUILDX_BAKE_HCL_FILE}" integration-tests
+
+.PHONY: build-local-infrastructure
```

Comment:
Can you move build-local-infrastructure above the build-test-* targets that way its easier to find the command?